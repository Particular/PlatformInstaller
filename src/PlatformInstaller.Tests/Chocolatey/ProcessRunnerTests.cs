using System;
using NUnit.Framework;
using ObjectApproval;

[TestFixture]
[Explicit]
public class ProcessRunnerTests
{

    [Test]
    public void VerifyOutputReceived()
    {
        var processRunner = new ProcessRunner();
        string output = null;
        processRunner.RunProcess("ping", "localhost", s => output += s, s => output += s).Wait();
        ObjectApprover.VerifyWithJson(output.ReplaceCaseless(Environment.MachineName, ""));
    }

    [Test]
    public void VerifyErrorReceived()
    {
        var processRunner = new ProcessRunner();
        string output = null;
        processRunner.RunProcess("net", " use foo", s => output += s, s => output += s).Wait();
        ObjectApprover.VerifyWithJson(output.ReplaceCaseless(Environment.MachineName, ""));
    }

}