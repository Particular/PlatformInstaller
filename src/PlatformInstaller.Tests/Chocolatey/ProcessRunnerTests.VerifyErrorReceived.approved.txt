[
    "System error 67 has occurred.",
    "",
    "The network name cannot be found.",
    ""
]